crates/bevy_reflect/src/struct_trait.rs

```diff
@@ -508,6 +508,44 @@ impl Debug for DynamicStruct {
     }
 }
 
+impl FromIterator<(String, Box<dyn Reflect>)> for DynamicStruct {
```

Comment:
Nit: Maybe add a doc comment on the methods or impl blocks for these? Just to clarify that `String` is meant to be the name of the field.